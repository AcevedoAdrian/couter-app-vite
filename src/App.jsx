// Esta funciÃ³n no se va a volver a cargar en memoria cuando se realice el render de nuevo. Ya que esta fuera del functional component que hace el rende
const sum = (a, b) => {
  const c = a + b;
  return `El resultado es ${c}`;
};

export const App = () => {
  return (
    <>
      {sum(3, 3)}
      <h1>Hello World</h1>
    </>
  );
};
