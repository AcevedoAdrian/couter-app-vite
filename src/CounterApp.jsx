import { useState } from "react";
import PropTypes from "prop-types";

export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);

  const handleSum = (e) => {
    setCounter(counter + 1);
  };
  const handleSubtract = (e) => {
    setCounter(counter - 1);
  };

  const handleReset = (e) => {
    setCounter(value);
  };

  return (
    <>
      <h1>I am the component CounterApp </h1>
      <h2> {counter} </h2>
      <button onClick={handleSum}>+1</button>
      <button onClick={handleSubtract}>-1</button>
      <button onClick={handleReset}>Reset</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};

CounterApp.defaultProps = {
  value: 32,
};
