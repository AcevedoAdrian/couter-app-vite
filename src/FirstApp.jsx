import PropTypes from "prop-types";

export const FirstApp = ({ title, subTitle }) => {
  if (!title) {
    throw new Error("The title is null");
  }

  return (
    <>
      <h2 data-testid="test-title">{title}</h2>
      <p>{subTitle}</p>
      <p>{subTitle}</p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
};

FirstApp.defaultProps = {
  name: "Acevedo Adrian",
  subTitle: "Esto es un subtitulo",
  // title: "Esto es un Titulo",
};
