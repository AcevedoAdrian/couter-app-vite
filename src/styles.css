head,
body {
  background-color: #080808;
  color: #ffffff;
  font-family: Helvetica, Arial, sans-serif;
  font-size: 1.3rem;
  padding: 70px;
}

button {
  padding: 10px;
  font-size: 1.3rem;
  margin-right: 5px;
}
