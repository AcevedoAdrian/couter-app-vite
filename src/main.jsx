import React from "react";
import ReactDOM from "react-dom/client";
import { App } from "./App";
import { CounterApp } from "./CounterApp";
import { FirstApp } from "./FirstApp";
import "./styles.css";
ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    {/* <App /> */}
    {/* <FirstApp subTitle="About My" /> */}
    <CounterApp value={10} />
  </React.StrictMode>
);
