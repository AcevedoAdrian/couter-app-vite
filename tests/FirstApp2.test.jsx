import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe("Pruebas en <FirstApp /> pero con menos codigo y mas limpio", () => {
  const title = "Hola, Soy Adrian";
  const subTitle = "Hola soy un subtitulo";
  test("debe de hace match con el snapshot", () => {
    const { container } = render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
  });

  test("debe de mostra el mensaje 'Hola, Soy Adrian'", () => {
    render(<FirstApp title={title} />);
    expect(screen.getByText(title)).toBeTruthy();
    // expect(screen.getByText(title)).not.toBeTruthy();
  });

  test("debe de mostrar el titulo en h1", () => {
    render(<FirstApp title={title} />);
    expect(screen.getByRole("heading", { level: 2 }).innerHTML).toContain(
      title
    );
  });
  
  test("debe de mostrar los subtitulos enviados por props", () => {
    render(<FirstApp title={title} subTitle={subTitle} />);
    expect(screen.getAllByText(subTitle).length).toBe(2);
  });
});
