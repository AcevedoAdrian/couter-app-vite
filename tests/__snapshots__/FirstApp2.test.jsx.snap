// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> pero con menos codigo y mas limpio debe de hace match con el snapshot 1`] = `
<div>
  <h2
    data-testid="test-title"
  >
    Hola, Soy Adrian
  </h2>
  <p>
    Esto es un subtitulo
  </p>
  <p>
    Esto es un subtitulo
  </p>
</div>
`;
